prov/gni/src/gnix_ep.c

```diff
@@ -1054,6 +1054,44 @@ static void __ep_destruct(void *obj)
 
 	GNIX_TRACE(FI_LOG_EP_CTRL, "\n");
 
+	/*
+	 * complete any outstanding fab reqs
+	 * on this EP
+	 */
+	while (atomic_get(&ep->active_fab_reqs) != 0) {
+		ret = _gnix_nic_progress(ep->nic);
+		if (ret != GNI_RC_SUCCESS)
+			GNIX_WARN(FI_LOG_EP_CTRL,
+				"_gnix_nic_progress returned %s\n",
+				  fi_strerror(-ret));
+	}
```

Comment:
Is it possible that this could be an infinite loop?  If so we might we want to eventually time out here in some fashion (either time or number of tries).